### PR TITLE
Added Studio docs for email sign up and custom Gitlab server support

### PR DESCRIPTION
**Pull request by tapadipti**
**Main pages to review**

Changed:
- https://dvc-org-studio-accounts-vedlrg.herokuapp.com/doc/studio/get-started
- https://dvc-org-studio-accounts-vedlrg.herokuapp.com/doc/studio/user-guide/views/create-view
- https://dvc-org-studio-accounts-vedlrg.herokuapp.com/doc/studio/user-guide/teams
- https://dvc-org-studio-accounts-vedlrg.herokuapp.com/doc/studio/user-guide/account-management

New:
- https://dvc-org-studio-accounts-vedlrg.herokuapp.com/doc/studio/user-guide/connect-custom-gitlab-server

### CODE REVIEW

**Comment by tapadipti** [comment]
**This PR is ready for review. But I've marked it as `Draft` so that we don't merge it before the corresponding changes in the Studio application are shipped.**

**Review by amritghimire — APPROVED** [review]

File: content/docs/studio/user-guide/views/create-view.md
Comment (iesahin) [inline content/docs/studio/user-guide/views/create-view.md:18]: `<admon>`

File: content/docs/studio/user-guide/views/create-view.md
Comment (iesahin) [inline content/docs/studio/user-guide/views/create-view.md:27]: tabs can be used here as well.

File: content/docs/studio/get-started.md
Comment (tapadipti) [reply content/docs/studio/get-started.md:43]: There are special instructions for Bitbucket, and nor for GitLab.com. The special instructions are only for GitHub and for self-hosted GitLab. For this reason (because only a subset of all cases have the special instructions), I'm not creating tabs. I've changed them to `<admon>`.

File: content/docs/studio/user-guide/account-management.md
Comment (tapadipti) [reply content/docs/studio/user-guide/account-management.md:12]: How about `Git cloud`?

File: content/docs/studio/user-guide/account-management.md
Comment (tapadipti) [reply content/docs/studio/user-guide/account-management.md:20]: How about `Git cloud`?

File: content/docs/studio/user-guide/account-management.md
Comment (tapadipti) [reply content/docs/studio/user-guide/account-management.md:28]: I changed the content a bit. How is this: `You can add multiple email addresses to a single Studio account. You can login to the account with any of your verified email addresses.`?

File: content/docs/studio/user-guide/connect-custom-gitlab-server.md
Comment (tapadipti) [reply content/docs/studio/user-guide/connect-custom-gitlab-server.md:12]: Changed it to `This URL should either be publicly accessible, or, if Studio is deployed on-premises in a private network, then it could also be a URL in the same private network so that Studio can access it.`

File: content/docs/studio/user-guide/views/create-view.md
Comment (tapadipti) [reply content/docs/studio/user-guide/views/create-view.md:27]: Same as above: There are special instructions for Bitbucket, and nor for GitLab.com. The special instructions are only for GitHub and for self-hosted GitLab. For this reason (because only a subset of all cases have the special instructions), I'm not creating tabs. I've changed them to <admon>.

File: content/docs/studio/get-started.md
Comment (tapadipti) [reply content/docs/studio/get-started.md]: Done

File: content/docs/studio/get-started.md
Comment (tapadipti) [reply content/docs/studio/get-started.md]: Done

File: content/docs/studio/get-started.md
Comment (tapadipti) [reply content/docs/studio/get-started.md]: Done

File: content/docs/studio/get-started.md
Comment (tapadipti) [reply content/docs/studio/get-started.md]: Done

File: content/docs/studio/user-guide/account-management.md
Comment (tapadipti) [reply content/docs/studio/user-guide/account-management.md]: Done

File: content/docs/studio/user-guide/account-management.md
Comment (tapadipti) [reply content/docs/studio/user-guide/account-management.md]: Done

File: content/docs/studio/user-guide/connect-custom-gitlab-server.md
Comment (tapadipti) [reply content/docs/studio/user-guide/connect-custom-gitlab-server.md]: Done

File: content/docs/studio/user-guide/views/create-view.md
Comment (tapadipti) [reply content/docs/studio/user-guide/views/create-view.md]: Done

File: content/docs/studio/user-guide/views/create-view.md
Comment (tapadipti) [reply content/docs/studio/user-guide/views/create-view.md]: Done

File: content/docs/studio/user-guide/views/create-view.md
Comment (tapadipti) [reply content/docs/studio/user-guide/views/create-view.md:19]: Done

**Comment by tapadipti** [comment]
> I made an initial review. Thank you @tapadipti

Thank you @iesahin . I've addressed your comments above.

**Comment by jorgeorpinel** [comment]
Is this still a draft or ready for final review? Thanks

File: content/docs/studio/get-started.md
Comment (jorgeorpinel) [inline content/docs/studio/get-started.md:37]: 💅🏼  This whole block should be indented 2 spaces so it's part of the list item.

File: content/docs/studio/user-guide/account-management.md
Comment (jorgeorpinel) [reply content/docs/studio/user-guide/account-management.md]: I'm not sure these services are considered clouds. I wouldn't use that term here.

File: content/docs/studio/user-guide/account-management.md
Comment (jorgeorpinel) [reply content/docs/studio/user-guide/account-management.md]: "Git hosting account" ?

File: content/docs/studio/user-guide/connect-custom-gitlab-server.md
Comment (jorgeorpinel) [reply content/docs/studio/user-guide/connect-custom-gitlab-server.md]: How about

> A publicly accessible URL for your GitLab server, or a URL in the same private network if Studio is deployed on-premises.

File: content/docs/studio/user-guide/account-management.md
Comment (tapadipti) [reply content/docs/studio/user-guide/account-management.md]: So you suggest keeping `your connected Git account`?

File: content/docs/studio/user-guide/account-management.md
Comment (tapadipti) [reply content/docs/studio/user-guide/account-management.md]: Done.

File: content/docs/studio/user-guide/connect-custom-gitlab-server.md
Comment (tapadipti) [reply content/docs/studio/user-guide/connect-custom-gitlab-server.md]: Changed to `A URL for your GitLab server. The URL should either be publicly accessible, or in the same private network as Studio if Studio is deployed on-premises.`

File: content/docs/studio/get-started.md
Comment (tapadipti) [reply content/docs/studio/get-started.md:37]: Done.

**Comment by tapadipti** [comment]
> Is this still a draft or ready for final review? Thanks

@jorgeorpinel It's ready for review. I've marked it as Draft so that we don't accidentally merge it before the corresponding changes in the Studio application are shipped.

**Comment by tapadipti** [comment]
@iesahin could you pls review and let me know if any more changes are needed? Merging this is blocked until you approve.

**Review by mvshmakov — APPROVED** [review]

**Review by iesahin — APPROVED** [review]
Thank you @tapadipti LGTM 👍🏼

**Comment by jorgeorpinel** [comment]
> I've marked it as Draft so that we don't accidentally merge it before the corresponding changes in the Studio application are shipped

For the record we use the status: wait-core-merge label for that 🙂 (drafts can be merged accidentally anyway).

And thanks! Great stuff as usual

File: content/docs/studio/get-started.md
Comment (jorgeorpinel) [inline content/docs/studio/get-started.md:9]: Not sure we needed the .com/org s but OK 🙂

File: content/docs/studio/get-started.md
Comment (jorgeorpinel) [inline content/docs/studio/get-started.md:17]: Pretty standard. Prob no admonition needed.

File: content/docs/studio/user-guide/account-management.md
Comment (jorgeorpinel) [inline content/docs/studio/user-guide/account-management.md:11]: OK all the .coms and .orgs start to get repetitive after a bit. I'd ✂️ them (minor).

File: content/docs/studio/user-guide/account-management.md
Comment (jorgeorpinel) [inline content/docs/studio/user-guide/account-management.md:27]: > You can add multiple email addresses to a single Studio account. You can login to the account with any of your verified email addresses.

May I ask why multiple addresses is allowed?

And furthermore why allowing login with any of them? And is that true even if the signup was via GH/GL/BB? (unclear from this doc)

File: content/docs/studio/user-guide/account-management.md
Comment (jorgeorpinel) [inline content/docs/studio/user-guide/account-management.md:43]: provider -> hosting?

(Consistent terminology vs other pages)

File: content/docs/studio/user-guide/account-management.md
Comment (jorgeorpinel) [inline content/docs/studio/user-guide/account-management.md:44]: > to grant access to your account to Iterative Studio

-> to grant Iterative Studio access to your account

File: content/docs/studio/user-guide/account-management.md
Comment (jorgeorpinel) [inline content/docs/studio/user-guide/account-management.md:45]: The last sentence is redundant, I think.

File: content/docs/studio/user-guide/account-management.md
Comment (jorgeorpinel) [inline content/docs/studio/user-guide/account-management.md:49]: More copy edits:
```suggestion
  If you signed up to use Studio with an email address, you will not have any
  Git connections by default. You can set them in this section.
```

File: content/docs/studio/user-guide/account-management.md
Comment (jorgeorpinel) [inline content/docs/studio/user-guide/account-management.md:55]:
```suggestion
  should create a team and
  [set up the GitLab server connection](/doc/studio/user-guide/connect-custom-gitlab-server)
  in the team settings.
```
> Needs wrapping

File: content/docs/studio/user-guide/account-management.md
Comment (jorgeorpinel) [inline content/docs/studio/user-guide/account-management.md:57]:
```suggestion
- Disconnect from your GitHub, GitLab or Bitbucket account.

```

File: content/docs/studio/user-guide/connect-custom-gitlab-server.md
Comment (jorgeorpinel) [inline content/docs/studio/user-guide/connect-custom-gitlab-server.md:17]: Maybe link to GitLab docs that refer to this instead of providing a URL (that could change in the future)

File: content/docs/studio/user-guide/teams.md
Comment (jorgeorpinel) [inline content/docs/studio/user-guide/teams.md:126]: Shouldn't need md bolding in headers. Need a change of styles?

File: content/docs/studio/get-started.md
Comment (tapadipti) [reply content/docs/studio/get-started.md:9]: I added these to distinguish them from on-prem/custom GitLab (and maybe GitHub in the future).

File: content/docs/studio/get-started.md
Comment (tapadipti) [reply content/docs/studio/get-started.md:17]: Changed.

File: content/docs/studio/user-guide/account-management.md
Comment (tapadipti) [reply content/docs/studio/user-guide/account-management.md:11]: See previous comment for why I added them.

File: content/docs/studio/user-guide/account-management.md
Comment (tapadipti) [reply content/docs/studio/user-guide/account-management.md:27]: Multiple email addresses are allowed, just like in GitHub. The primary one can't be deleted; it's the one that gets account notifications. You could change your email address for a Studio account by adding a new one and setting that as primary and then deleting the old (non-primary) one.

If they are all my email addresses, they why not let me log in with any of them? Yes, it's true regardless of the signup method, as long as you have set a password in your account - have added this to the doc.

File: content/docs/studio/user-guide/account-management.md
Comment (tapadipti) [reply content/docs/studio/user-guide/account-management.md:43]: Changed it to `Git hosting provider`

File: content/docs/studio/user-guide/account-management.md
Comment (tapadipti) [reply content/docs/studio/user-guide/account-management.md:44]: Done.

File: content/docs/studio/user-guide/account-management.md
Comment (tapadipti) [reply content/docs/studio/user-guide/account-management.md:45]: Removed.

File: content/docs/studio/user-guide/account-management.md
Comment (tapadipti) [reply content/docs/studio/user-guide/account-management.md:49]: Changed. Minor edit -> Changed `You can set them in this section.` to `You can set them up in this section.`

File: content/docs/studio/user-guide/account-management.md
Comment (tapadipti) [reply content/docs/studio/user-guide/account-management.md:57]: There's a reason the `.com/org`s were addded (see previous comments). But I guess the user will understand here, so I've removed then as you suggested. But I didn't change `accounts` to `account`, coz the user can disconnect from multiple Git accounts (each on a different provider).

File: content/docs/studio/user-guide/connect-custom-gitlab-server.md
Comment (tapadipti) [reply content/docs/studio/user-guide/connect-custom-gitlab-server.md:17]: This URL is used in our code as well. So, if/when this changes, we will have to update Studio anyway.

File: content/docs/studio/user-guide/connect-custom-gitlab-server.md
Comment (tapadipti) [reply content/docs/studio/user-guide/connect-custom-gitlab-server.md:41]: - Change the second and third instances to code (`gitlab.com`) instead of link.
- Yes, we need `.com`, coz we are trying to distinguish between `gitlab.com` and self-managed GitLab instances.
- I've used GitLab (G and L in caps) everywhere. Let me know if you see any discrepancy. Also, in this block, I had `GitLab.com`, which I have now corrected to `gitlab.com` (coz it's the URL and not the name).

File: content/docs/studio/user-guide/teams.md
Comment (tapadipti) [reply content/docs/studio/user-guide/teams.md:126]: Since this is a 4th level heading, it's almost not distinguishable from the rest of the text. That's why I had to make it bold, so that it looks at least somewhat different (as shown below).

image

File: content/docs/studio/user-guide/teams.md
Comment (tapadipti) [reply content/docs/studio/user-guide/teams.md:132]: Done.

**Comment by tapadipti** [comment]
@jorgeorpinel I've made your suggested changes in a new PR: https://github.com/iterative/dvc.org/pull/3497 And for suggestions that I've not implemented, pls see the comment replies above. Let me know if more changes are needed.

File: content/docs/studio/get-started.md
Comment (jorgeorpinel) [reply content/docs/studio/get-started.md:9]: I see.

File: content/docs/studio/user-guide/account-management.md
Comment (jorgeorpinel) [reply content/docs/studio/user-guide/account-management.md:27]: > Multiple email addresses are allowed, just like in GitHub

Yes but why? Is there a use case? I'm just curious -- sounds like possibly unnecessary complication.

File: content/docs/studio/user-guide/account-management.md
Comment (jorgeorpinel) [reply content/docs/studio/user-guide/account-management.md:43]: Thanks

File: content/docs/studio/user-guide/connect-custom-gitlab-server.md
Comment (jorgeorpinel) [reply content/docs/studio/user-guide/connect-custom-gitlab-server.md:41]: gitlab.com is OK for links. Not sure we need code blocks with the URL. Anyway, pretty minor (I left a suggestion in the follow-up PR.

File: content/docs/studio/user-guide/teams.md
Comment (jorgeorpinel) [reply content/docs/studio/user-guide/teams.md:126]: Need a change of styles? 🙂

File: content/docs/studio/user-guide/account-management.md
Comment (tapadipti) [reply content/docs/studio/user-guide/account-management.md:27]: Someone could use either of their email addresses to login (if they have multiple - such as for work and personal). It also lets them change the email address for their Studio account by adding the new one and deleting the old one (although for this, we could have added a `Change email` functionality). Also, I don't remember, but I think there was another (more technical) reason for this that applied to email signups. I can ask in the team if necessary.

Comment (iesahin) [reply content/docs/studio/user-guide/account-management.md:27]: It's possible to lose access to email accounts for some reason and keeping multiple email addresses recorded in services (if possible) is usually a better approach. @jorgeorpinel

**Pull request merged**
